pkg/planner/core/indexmerge_path.go

```diff
@@ -596,7 +930,191 @@ func (ds *DataSource) generateIndexMergeOnDNF4MVIndex(normalPathCnt int, filters
 	return
 }
 
-// generateIndexMergeJSONMVIndexPath generates paths for (json_member_of / json_overlaps / json_contains) on multi-valued index.
+// generateIndexMerge4ComposedIndex generates index path composed of multi indexes including multivalued index from
+// (json_member_of / json_overlaps / json_contains) and single-valued index from normal indexes.
+/*
+CNF path
+	1. select * from t where ((1 member of (a) and c=1) and (2 member of (b) and d=2) and (other index predicates))
+		flatten as: select * from t where 1 member of (a) and 2 member of (b) and c=1 and c=2 and other index predicates
+		analyze: find and utilize index access filter items as much as possible:
+		IndexMerge(AND-INTERSECTION)                  --- ROOT
+			IndexRangeScan(mv-index-a)(1)             --- COP   ---> expand as: IndexMerge(AND)          ---> simplify: since memberof only have 1 partial index plan, we can defer the TableRowIdScan(t).
+			IndexRangeScan(mv-index-b)(2)             --- COP						IndexRangeScan(a, [1,1])
+			IndexRangeScan(non-mv-index-if-any)(?)    --- COP		        		TableRowIdScan(t)
+			Selection(remained-non-index-predicates)  --- COP
+				TableRowIdScan(t)  					  --- COP
+	2. select * from t where ((1 member of (a) and c=1) and (json_contains(b, '[1, 2, 3]') and d=2) and (other index predicates))
+		flatten as: select * from t where 1 member of (a) and json_contains(b, '[1, 2, 3]') and c=1 and c=2 and other index predicates
+		analyze: find and utilize index access filter items as much as possible:
+		IndexMerge(AND-INTERSECTION)                  --- ROOT
+			IndexRangeScan(mv-index-a)(1)             --- COP
+			IndexMerge(mv-index-b AND-INTERSECTION)   --- ROOT (embedded index merge) ---> simplify: we can defer the TableRowIdScan(t) to the outer index merge.
+				IndexRangeScan(a, [1,1])              --- COP
+				IndexRangeScan(a, [2,2])              --- COP
+				IndexRangeScan(a, [3,3])              --- COP
+			IndexRangeScan(non-mv-index-if-any)(?)    --- COP
+			TableRowIdScan(t)                         --- COP
```

Comment:
Got a bit confused by those nested IndexMerge and "simplify" descriptions. For me, I think they make the comments harder to understand.